src/table/table.stories.ts

```diff
@@ -0,0 +1,268 @@
+import {
+	TemplateRef,
+	Component,
+	ViewChild,
+	OnInit,
+	Input
+} from "@angular/core";
+import { storiesOf, moduleMetadata } from "@storybook/angular";
+import {
+	withKnobs,
+	boolean,
+	selectV2
+} from "@storybook/addon-knobs/angular";
+
+import { TranslateModule } from "@ngx-translate/core";
+
+import {
+	TableModule,
+	TableModel,
+	TableItem,
+	TableHeaderItem,
+	NFormsModule
+} from "../";
+
+@Component({
+	selector: "app-custom-table",
+	template: `
+		<button class="bx--btn bx--btn--sm bx--btn--primary" (click)="addRow()">Add row</button>
+		<button class="bx--btn bx--btn--sm bx--btn--primary" (click)="addColumn()">Add column</button>
+
+		<ng-template #customTableItemTemplate let-data="data">
+			<a [attr.href]="data.link">{{data.name}} {{data.surname}}</a>
+		</ng-template>
+		<ng-template #customHeaderTemplate let-data="data">
+			<i><a [attr.href]="data.link">{{data.name}}</a></i>
+		</ng-template>
+
+		<ibm-table
+			[model]="model"
+			[size]="size"
+			[showSelectionColumn]="showSelectionColumn"
+			[striped]="striped"
+			(sort)="customSort($event)">
+		</ibm-table>
+	`
+})
+class DynamicTableStory implements OnInit {
+	@Input() model = new TableModel();
+	@Input() size = "md";
+	@Input() showSelectionColumn = true;
+	@Input() striped = true;
+
+	@ViewChild("customHeaderTemplate")
+	private customHeaderTemplate: TemplateRef<any>;
+	@ViewChild("customTableItemTemplate")
+	private customTableItemTemplate: TemplateRef<any>;
+
+	ngOnInit() {
+		this.model.data = [
+			[new TableItem({data: "Name 1"}), new TableItem({data: {name: "Lessy", link: "/table"}, template: this.customTableItemTemplate})],
+			[new TableItem({data: "Name 3"}), new TableItem({data: "swer"})],
+			[new TableItem({data: "Name 2"}), new TableItem({data: {name: "Alice", surname: "Bob"}, template: this.customTableItemTemplate})],
+			[new TableItem({data: "Name 4"}), new TableItem({data: "twer"})],
+		];
+		this.model.header = [
+			new TableHeaderItem({data: "Very long title indeed"}),
+			new CustomHeaderItem({
+				data: {name: "Custom header", link: "/table"},
+				template: this.customHeaderTemplate,
+				style: {"width": "auto"}
+			})
+		];
+	}
+
+	customSort(index: number) {
+		this.sort(this.model, index);
+	}
+
+	sort(model, index: number) {
+		if (model.header[index].sorted) {
+			// if already sorted flip sorting direction
+			model.header[index].ascending = model.header[index].descending;
+		}
+		model.sort(index);
+	}
+
+	addRow() {
+		const lastRowCopy = JSON.parse(JSON.stringify(this.model.row(this.model.data.length - 1)));
```

Comment:
Should we move this into a Tools file?

This is what I'm doing in charts
![image](https://user-images.githubusercontent.com/14989804/44595033-dc224800-a795-11e8-824a-a6de31d3a345.png)